File: Tests/Reqnroll.Formatters.Tests/Samples/Resources/global-hooks/global_hooks.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Reqnroll;

namespace Reqnroll.Formatters.Tests.Samples.Resources.global_hooks;

[Binding]
public class global_hooks
{
    [BeforeTestRun]
    public static void BeforeTestRun1()
    {

    }

    [BeforeTestRun]
    public static void BeforeTestRun2()
    {

    }

    [When("a step passes")]
    public void AStepPasses()
    {

    }

    [When("a step fails")]
    public void AStepFails()
    {
        throw new ApplicationException("Exception in step");
    }

    [AfterTestRun]
    public static void AfterTestRun1()
    {

    }
    [AfterTestRun]
    public static void AfterTestRun2()
    {

    }
}

File: Tests/Reqnroll.Formatters.Tests/Samples/Resources/global-hooks/global_hooks.ts
import { When, BeforeAll, AfterAll } from '@cucumber/fake-cucumber'

BeforeAll({}, function () {
  // no-op
})

BeforeAll({}, function () {
  // no-op
})

When('a step passes', function () {
  // no-op
})

When('a step fails', function () {
  throw new Error('Exception in step')
})

AfterAll({}, function () {
  // no-op
})

AfterAll({}, function () {
  // no-op
})
